tests/pytorch/distributed/run_layer_with_overlap.py

```diff
@@ -210,6 +267,11 @@ def _train(opts):
         opts.tcp_init = True
         opts.bind_to_device = True
         opts.bootstrap_backend = "mpi"
+    elif "SLURM_PROCID" in os.environ:
+        WORLD_RANK = (int)(os.environ["SLURM_PROCID"])
+        LOCAL_RANK = (int)(os.environ["SLURM_LOCALID"])
+        WORLD_SIZE = (int)(os.environ["SLURM_NTASKS"])
+        LOCAL_SIZE = WORLD_SIZE
```

Comment:
This breaks our CI pipelines. On SLURM systems we allocate a single node with `srun -N 1` (so `SLURM_NTASKS=1`) and then start distributed sessions with `torchrun`. When I look at the logs, it seems like we are just running these tests without any parallelism.
```suggestion
```
If we want to keep support for SLURM envvars, we should put them after the `torchrun` envvars.